tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,162 @@
+    if running_status:
+        duthost.shell("docker exec {} kill -9 {}".format(container_name, process_id)) 
+    else:
+        pytest_assert(False, "Failed to find {} process in {}".format(process, container_name))
+
+    time.sleep(7)
+
+    running_status, process_id = get_process_id(duthost, container_name, process)
+    if running_status:
+        pytest_assert(False, "Failed to stop {} process before test".format(process))
+    else:
+        logging.info("{} process in {} is stopped successfully".format(process, container_name)) 
+
+def check_container_status(duthost, container_name, should_be_stopped):
+    """
+    @summary: Determine whether a container should be in running state or not
+    """
+    is_running = duthost.shell("docker inspect -f \{\{.State.Running\}\} %s" % container_name)
+    if is_running['stdout'].strip() == "false" and should_be_stopped:
+        return True
+    if is_running['stdout'].strip() == "true" and not should_be_stopped:
+        return True
+    return False
+ 
+def verify_autorestart_with_critical_process(duthost, container_name, process):
+    """
+    @summary: Killing a critical process in a container to verify whether the container 
+              can be stopped and then restarted correctly 
+    """
+    kill_process(duthost, container_name, process)
+
+    logging.info("Waiting until {} is stopped...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, True), 
+                  "Failed to stop {}".format(container_name))
+    logging.info("{} is stopped".format(container_name))
+
+    logging.info("Waiting until {} is restarted...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_RESTART_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False),
+                  "Failed to restart {}".format(container_name))
+    logging.info("{} is restarted".format(container_name))
+
+def verify_no_autorestart_with_non_critical_process(duthost, container_name, process):
+    """
+    @summary: Killing a non-critical process in a container to verify whether the container 
+              is still in the running state 
+    """
+    kill_process(duthost, container_name, process)
+
+    logging.info("Checking whether the {} is still running...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False),
+                  "{} is stopped unexpectedly".format(container_name))
+    logging.info("{} is running".format(container_name))
+
+@pytest.fixture(scope="module", autouse=True)
+def change_autorestart_state(duthost):
```

Comment:
This function name is vague. This function specifically **enables** the autorestart feature for all supported containers.